### DISCUSSION TITLE
Add type field in PG-JSONL

### DISCUSSION THREAD

**Discussion by nichtich** [pg-format/pg-formatter#35] [Q&A] [answered]
Processing PG-JSONL may be a little bit easier if each record hat a `type` field with value `node` or `edge`.

~~~
{"type":"node","id":"101","labels":["person"],"properties":{"name":["Alice","Carol"],"country":["United States"]}}
{"type":"node","id":"102","labels":["person","student"],"properties":{"name":["Bob"],"country":["Japan"]}}
{"type":"node","from":"101","to":"102","labels":["same_school","same_class"],"properties":{"since":[2012]},"undirected":true}
{"type":"edge","from":"101","to":"102","labels":["likes"],"properties":{"since":[2015],"engaged":[false]}}
~~~

The current data model does not require this but as soon as extensions such as edge identifiers and graph attributes are possible, the detection of record type will get complicated. 

Pros:
- Simplify processing (e.g. filter nodes with `jq 'select(.type=="node")'`
- Allow for extensions

Cons:
- redundant

**Comment by hchiba1** [comment]
I see, it is a reasonable extension.
Question: Is the `type` field optional or mandatory?

**Reply by nichtich** [reply, marked as answer]
Mandatory would be more robust for data consumers. The documentation could state that applications SHOULD also accept PG-JSONL without `type` but MUST include the field when generating PG-JSONL.
